lib/project_types/extension/models/specification_handlers/default.rb

```diff
@@ -42,12 +42,14 @@ def valid_extension_contexts
           []
         end
 
-        def choose_port?
-          false
+        def choose_port?(context)
+          renderer_package = renderer_package(context)
+          Features::ArgoCliCompatibility.new(context: context, renderer_package: renderer_package).can_choose_port?
         end
 
-        def establish_tunnel?
-          false
+        def establish_tunnel?(context)
+          renderer_package = renderer_package(context)
+          Features::ArgoCliCompatibility.new(context: context, renderer_package: renderer_package).can_establish_tunnel?
```

Comment:
this could be cleaned up in the future if we move this outside of the `default` handler and into an `argo_admin` handler.